rpc/ws/client.go

```diff
@@ -54,6 +54,14 @@ const (
 
 // Connect creates a new websocket client connecting to the provided endpoint.
 func Connect(ctx context.Context, rpcEndpoint string) (c *Client, err error) {
+	return ConnectWithOptions(ctx, rpcEndpoint, nil)
+}
+
+// ConnectWithOptions creates a new websocket client connecting to the provided
+// endpoint with a http header if available The http header can be helpful to
+// pass basic authentication params as prescribed
+// ref https://github.com/gorilla/websocket/issues/209
+func ConnectWithOptions(ctx context.Context, rpcEndpoint string, requestHeader http.Header) (c *Client, err error) {
```

Comment:
hey @mario-zh 

thanks for the PR!

how about making this a general `Options` struct? that way more options could be added in the future.